src/rpp/rpp/operators/combine_latest.hpp

```diff
@@ -0,0 +1,166 @@
+
+    /**
+     * \brief The coordinator that caches the values from all the observables and dispatches latest caches to the observer. Note the emission is only sent to the observer when all the observables at least emits once.
+     */
+    template<constraint::decayed_type SourceValueType, constraint::decayed_type ...OtherValueTypes>
+    struct combine_latest_coordinator_t
+    {
+        explicit combine_latest_coordinator_t(TCombiner combiner) : m_combiner(combiner) {}
+
+        // Not copyable nor movable.
+        combine_latest_coordinator_t(const combine_latest_coordinator_t&) = delete;
+        combine_latest_coordinator_t(combine_latest_coordinator_t&&) = delete;
+
+        template<size_t I>
+        void offer_next_value(auto&& value, const auto& subscriber)
+        {
+            if (!subscriber.is_subscribed()) return;
+
+            std::scoped_lock<std::mutex> lock{m_mutex};
+            std::get<I>(m_values) = std::forward<decltype(value)>(value);
+
+            // If all the optional(s) have value, dispatch to downstream
+            if (has_values_for_observer())
+            {
+                subscriber.on_next(combine_cached_values());
+            }
+        }
+
+        void offer_completed(const auto& subscriber)
+        {
+            if (!subscriber.is_subscribed()) return;
+
+            std::scoped_lock<std::mutex> lock{m_mutex};
+
+            if (++m_completed_count == m_total_completed_count)
+            {
+                subscriber.on_completed();
+            }
+        }
+
+    private:
+        TCombiner m_combiner;
+
+        std::mutex m_mutex;
+        std::tuple<std::optional<SourceValueType>, std::optional<OtherValueTypes>...> m_values{};
+        size_t m_completed_count{0};
+
+        const std::size_t m_total_completed_count = 1 + sizeof...(OtherValueTypes);
+
+        bool has_values_for_observer()
+        {
+            return std::apply([](const auto&... cached_values)
+            {
+                // Variadic expansion used
+                return (cached_values.has_value() && ...);
+            }, m_values);
+        }
+
+        auto combine_cached_values()
+        {
+            return std::apply([&](const auto&... cached_values)
+            {
+                // Variadic expansion used
+                return m_combiner(cached_values.value()...);
+            }, m_values);
+        }
+    };
+
+    /**
+     * \brief Templated helper function for subscribing to variadic 'other' observables.
+     *
+     * \param observables is a variadic tuple. The first observable is the current observable and rest are the other observables.
+     * \param subscriber is the downstream subscriber.
+     * \param coordinator manages the cache of emission from the observables and coordinate dispatching.
+     */
+    template<size_t...I>
+    static void subscribe_other_observables(std::index_sequence<I...>, // Used in compile time for variadic expansion
```

Comment:
Why static? we pass m_other_observables as observables, but it can be just "member non-static function"